### PR TITLE
7.x 2.x spotlight dropdowns css attachment

### PR DESCRIPTION
**Pull request by cjwest**
# READY FOR REVIEW

# Summary
- This uses an attachment view to display the first spotlight.

- Available for review at:

https://sites.stanford.edu/jse-soe-dev/spotlight

# Needed By (Date)
- Sprint end

# Criticality
- Client is eagerly awaiting this.

# Steps to Test

# Affects 
- SoE (School of Engineering website)

# Associated Issues and/or People
## Related JIRA ticket(s)
https://stanfordits.atlassian.net/browse/SOE-2966
https://stanfordits.atlassian.net/browse/SOE-2976
https://stanfordits.atlassian.net/browse/SOE-2986
https://stanfordits.atlassian.net/browse/SOE-3002

## More Information - To migrate this working code:
1. Need the library on dev at default/libraries/history.js
2. Need the patched version of views_load_more on dev: /default/modules/contrib/views_load_more
3. Need the patched version of views_ajax_history on dev: /default/modules/contrib/views_ajax_history
4. Need to revert stanford_people_spotlight_views

## Folks to notify
@boznik 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: modules/stanford_people_spotlight_views/js/stanford_people_spotlight_views.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+(function ($) {
+  Drupal.behaviors.peopleSpotlightViews = {
+    attach: function (context, settings) {
+
+        function spotlightTransition() {
```
Comment (sherakama) [inline modules/stanford_people_spotlight_views/js/stanford_people_spotlight_views.js]: Needs more code comments. What is this thing doing?

File: modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module [hunk 2]
```diff
@@ -5,3 +5,43 @@
...
+ */
+function stanford_people_spotlight_views_form_alter(
+  &$form,
+  &$form_state,
+  $form_id) {
+
+  // Handle exposed filter dropdowns for spotlight page.
+  switch ($form['#id']) {
+    case "views-exposed-form-stanford-ppl-spot-3-v-card-page-1":
+      $form['field_s_ppl_spot_affiliation_tid']['#options']["All"] = 'All people';
```
Comment (sherakama) [inline modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module]: Text should be translated.

File: modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module [hunk 3]
```diff
@@ -5,3 +5,43 @@
...
+      $form['field_s_ppl_spot_department_tid']['#options']["All"] = 'All departments';
```
Comment (sherakama) [inline modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module]: Translate this too.

File: modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module [hunk 4]
```diff
@@ -5,3 +5,43 @@
  */
 
 include_once 'stanford_people_spotlight_views.features.inc';
+/**
+ * Implements template_preprocess_views_view()
+ *
+ * @link https://api.drupal.org/api/views/theme%21theme.inc/function/template_preprocess_views_view/7.x-3.x
+ */
+
```
Comment (sherakama) [inline modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module]: no space between comment and function.

File: modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module [hunk 5]
```diff
@@ -5,3 +5,43 @@
...
+function stanford_people_spotlight_views_preprocess_views_view(&$vars) {
+  // Get the current view info
+  $view = $vars['view'];
+
+  // Add CSS/JS based on view name
+  if ($view->name == 'stanford_ppl_spot_3_v_card') {
+    //Add CSS/JS based on current view display
+    if ($view->current_display == 'page_1') {
+      drupal_page_is_cacheable(FALSE);
```
Comment (sherakama) [inline modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module]: Why is the page being forced to no cache?

**Comment by boznik** [comment]
Thx @sherakama ..i'll fix this up.

File: modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module (hunk 5 above)
Comment (boznik) [reply modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module]: There was an issue with views_load_more and the caching due to the exposed filters

Comment (sherakama) [reply modules/stanford_people_spotlight_views/stanford_people_spotlight_views.module]: Ok, please leave code comments as to why. Help the next person out figure why the varnish hit rate is low on that page. 👍

**Comment by minorwm** [comment]
@boznik Looks good to me and works like a charm. Merging now.

**Pull request merged**